packages/@aws-cdk/pipelines/lib/stage.ts

```diff
@@ -224,6 +289,61 @@ export class CdkStage extends CoreConstruct {
     return stripPrefix(s, `${this.stageName}-`);
   }
 
+  /**
+   * Add a security check before the prepare/deploy actions of an CDK stage.
+   * The security check consists of two actions:
+   *  - CodeBuild Action to check for security changes in a stage
+   *  - Manual Approval Action that is auto approved via a Lambda if no security changes detected
+   */
+  private addSecurityCheck(appStageName: string, options?: BaseStageOptions) {
+    const { cdkDiffProject } = this.getApplicationSecurityCheck();
+    const notificationTopic: sns.ITopic | undefined = options?.securityNotificationTopic ?? this.securityNotificationTopic;
+    notificationTopic?.grantPublish(cdkDiffProject);
+
+    const approveActionName = `${appStageName}ManualApproval`;
+    const diffAction = new CodeBuildAction({
+      runOrder: this.nextSequentialRunOrder(),
+      actionName: `${appStageName}SecurityCheck`,
+      input: this.cloudAssemblyArtifact,
+      project: cdkDiffProject,
+      variablesNamespace: `${appStageName}SecurityCheck`,
+      environmentVariables: {
+        STACK_NAME: {
+          value: this.pipelineStage.pipeline.stack.stackName,
+          type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+        },
+        STAGE_NAME: {
+          value: this.stageName,
+          type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+        },
+        ACTION_NAME: {
+          value: approveActionName,
+          type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+        },
+        ...notificationTopic ? {
+          NOTIFICATION_ARN: {
+            value: notificationTopic.topicArn,
+            type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+          },
+          NOTIFICATION_SUBJECT: {
+            value: `Security Changes detected in ${appStageName}`,
```

Comment:
Try to make messages actionable. What should the user do when they receive this message?

(I would phrase it like "confirm broadened IAM permissions in deployment of ...")